sklearn/decomposition/pca.py

```diff
@@ -388,7 +388,7 @@ def _fit(self, X):
         else:
             n_components = self.n_components
 
-        # Handle svd_solver
+        # Handle svd_solver and _fit_method
         svd_solver = self.svd_solver
```

Comment:
We can replace variable ``svd_solver`` with the new ``self._fit_method``

Reply:
Everywhere? IMO `svd_solver` is more concise/readable..

Reply:
`svd_solver` is just being a local variable in `_fit()` method. If we replace it with a class variable (say `self._fit_svd_solver` for now), it will serve both the purposes, doing `svd_solver`'s current task and expose the solver used. 

Moreover, `NearestNeighbours` also handle `algorithm='auto'` in this manner.

Reply:
Yes, but you would have to do 4-5 replacements of `svd_solver` with `self._fit_svd_solver` to save one assignment, `self._fit_svd_solver = svd_solver` at the end (current solution). In any case that's not too important, either way would be fine, up to you.